code/drasil-lang/lib/Language/Drasil/Document.hs

```diff
@@ -86,8 +86,33 @@ checkToC (Notebook t a sc) = Notebook t a sc
 -- * Content Constructors
 
 -- | Smart constructor for labelled content chunks.
-llcc :: Reference -> RawContent -> LabelledContent
-llcc r = LblC (r ^. uid) r
+-- Now builds a Reference using the provided UID instead of extracting it from the Reference.
+llcc :: UID -> LblType -> ShortName -> RawContent -> LabelledContent
+llcc u lbl sn rc = LblC u (Reference u lbl sn) rc
+
+-- | Helper for creating labelled content with a figure reference.
+llccFig :: String -> RawContent -> LabelledContent
+llccFig rs rc = llcc (docUid rs) (RP (prepend "Fig") ("Figure:" ++ repUnd rs)) (shortname' (S rs)) rc
+
+-- | Helper for creating labelled content with a table reference.
+llccTab :: String -> RawContent -> LabelledContent
+llccTab rs rc = llcc (docUid rs) (RP (prepend "Tab") ("Table:" ++ repUnd rs)) (shortname' (S rs)) rc
+
+-- | Helper for creating labelled content with an equation reference.
+llccEqn :: String -> RawContent -> LabelledContent
+llccEqn rs rc = llcc (docUid rs) (RP (prepend "Eqn") ("Equation:" ++ repUnd rs)) (shortname' (S rs)) rc
+
+-- | Helper for creating labelled content with a UID-based figure reference.
+llccFig' :: UID -> RawContent -> LabelledContent
+llccFig' rs rc = llcc (docNs rs) (RP (prepend "Fig") ("Figure:" ++ repUnd (show rs))) (shortname' (S $ show rs)) rc
```

Comment:
1. This comment should be `Create a figure with a document-referrable label.` Follow this format for the other smart constructors created by this PR.
2. If `makeFigRef` is no longer used, it should be removed. Similar to other constructors for `Reference`.
3. `llccFig` should be moved to the same file containing the definition of `data LabelledContent`.
4. `llccFig` and related constructors should exclusively accept `UID`s (not `String`s) for the input/creation of the internal `UID`. Remove the duplicate constructors as a result of changing this.